doc/releases/changelog-dev.md

```diff
@@ -25,6 +25,10 @@
 * `AdaptiveOptimizer` is updated to use non-default user-defined qnode arguments.
   [(#3765)](https://github.com/PennyLaneAI/pennylane/pull/3765)
 
+* 3 new decomposition algorithms are added for n-controlled operations with single-qubit target,
+  and are selected automatically when they produce a better result.
```

Comment:
Awesome! It it always clear which decomposition is better? Are there any major classical computational overheads in each approach?

Reply:
Between these 3, yes. OD > MD > General ~ ZYZ. Based on earlier discussions the general bisect algorithm is almost always preferable over ZYZ since it uses less CNOTs. I think the only time ZYZ is chosen instead of the bisect algorithm is for n=1 since that is too small and a degenerate case for the bisect algorithm.

No, it's just some 2x2 matrix math, so it is quite fast to compute the decomposition.